### ISSUE TITLE
[Bug] `stg_zuora__order_tmp` error with Snowflake

### ISSUE THREAD

**Issue by raphaelvarieras** [fivetran/dbt_zuora#6]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

Snowflake doesn't like a table called `ORDER`, a reserved keyword. An error is generated upon building the package when using the Snowflake adapter.

### Relevant error log or model output

```shell
Database Error in model stg_zuora__order_tmp (models/tmp/stg_zuora__order_tmp.sql)
  001003 (42000): SQL compilation error:
  syntax error line 5 at position 27 unexpected 'order'.
  syntax error line 6 at position 2 unexpected ')'.
  compiled Code at target/run/zuora_source/models/tmp/stg_zuora__order_tmp.sql
```

### Expected behavior

The package documentation should indicate that Snowflake users need to add:
```
    zuora_order_identifier: "ORDER" 
```

And the package source file should be updated to use the quoting option:
```
      - name: order
        identifier: "{{ var('zuora_order_identifier', 'order')}}"
        description: "{{ doc('order_table') }}"
        quoting:
          identifier: true
```

### dbt Project configurations

N/A

### Package versions

```
  - package: fivetran/zuora
    version: 0.1.0
```

### What database are you using dbt with?

snowflake

### dbt Version

1.5

### Are you willing to open a PR to help address this issue?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @raphaelvarieras thanks for opening this issue and for addressing the issue in the above PR!

I am currently in the process of rolling your updates out to the source package. Once I do so, I will be sure to replicate your README changes within this package as well to ensure users are aware of how the package handles the `order` source for Snowflake users.

**Comment by fivetran-joemarkiewicz** [comment]
@raphaelvarieras thank you again for identifying and contributing these changes to the package! I have since merged these changes and cut a new v0.1.2 release of the zuora source package. You should see these changes live on the dbt hub at the top of the hour. 

Thank you again and please feel free to comment if the issue still persists on your end.
